src/browzine-summon-adapter.js

```diff
@@ -124,30 +131,63 @@ angular.module("summonApp.directives")
     return template;
   };
 
-  return {
-    link: function(scope, element, attributes) {
-      if(!shouldEnhance(scope)) {
-        return;
+  function documentSummary(scope, element) {
+    if(!shouldEnhance(scope)) {
+      return;
+    }
+
+    var endpoint = getEndpoint(scope);
+
+    $.getJSON(endpoint, function(response) {
+      var data = getData(response);
+      var browzineWebLink = getBrowZineWebLink(data);
+      var coverImageUrl = getCoverImageUrl(data, response);
+
+      if(browzineWebLink) {
+        var template = buildTemplate(data, browzineWebLink, bookIcon);
+        $(element).find(".docFooter .row:first").append(template);
       }
 
-      var endpoint = getEndpoint(scope);
+      if(coverImageUrl) {
+        $(element).find(".coverImage img").attr("src", coverImageUrl);
+      }
+    });
+  };
 
-      http.get(sce.trustAsResourceUrl(endpoint)).then(function(response) {
-        var data = getData(response);
-        var browzineWebLink = getBrowZineWebLink(data);
-        var coverImageUrl = getCoverImageUrl(data, response);
+  function browZineEnhance(element) {
+    var secret = Object.getOwnPropertyNames(element).filter(function(property) {
+      property = property.replace(/[^a-z]/gi, "");
+      return property.indexOf("jQuery") === 0;
+    });
 
-        if(browzineWebLink) {
-          var template = buildTemplate(data, browzineWebLink, bookIcon);
-          element.find(".docFooter .row:first").append(template);
-        }
+    var scope = element[secret].$scope;
 
-        if(coverImageUrl) {
-          element.find(".coverImage img").attr("src", coverImageUrl);
-        }
-      });
-    }
+    documentSummary(scope, element);
+  };
+
+  var results = document.querySelector("#results");
+  var config = {
+    attributes: true,
+    childList: true,
+    characterData: true,
+    subtree: true,
   };
-}]);
 
-// End BrowZine-Summon Integration Code
+  //Enhance any documentSummary elements present before the observer starts
+  var documentSummaries = results.querySelectorAll(".documentSummary");
+
+  Array.prototype.forEach.call(documentSummaries, function(documentSummary) {
```

Comment:
Does using `Array.prototype.forEach` mean we have more backward compatibility than if we did `documentSummaries.forEach` ?

Reply:
Yep. IE and older versions of Safari don't have a `forEach` on `querySelectorAll` results and I try to use jQuery for only DOM manipulation and AJAX convenience methods.